### PR TITLE
Fixed gofmt, ineffassign and misspell issues

### PR DESCRIPTION
**Pull request by jduranf**
Fixed all gofmt, ineffassign and misspell issues detected by the [report card](https://goreportcard.com/report/github.com/edgexfoundry/edgex-go).

Related with #501

Signed-off-by: Joan Duran <jduran@circutor.com>

### CODE REVIEW

File: internal/core/data/event.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func addNew(e models.Event) (string, error) {
 					return "", err
 				}
 			}
-			valid, err := isValidValueDescriptor(vd, e.Readings[reading])
+			valid, _ := isValidValueDescriptor(vd, e.Readings[reading])
```
Comment (feclare) [inline internal/core/data/event.go]: I know it is not directly related to this PR, but can you change the next if to add a test for err != nil

File: pkg/clients/coredata/reading_test.go [hunk 2]
```diff
@@ -17,8 +17,8 @@ package coredata
 
 import (
 	"fmt"
-	"github.com/edgexfoundry/edgex-go/pkg/clients/types"
 	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/pkg/clients/types"
```
Comment (feclare) [inline pkg/clients/coredata/reading_test.go]: Move up the net and testing imports

File: internal/core/data/event.go (hunk 1 above)
Comment (feclare) [reply internal/core/data/event.go]: Yes, I like your idea.

**Comment by jduranf** [comment]
@feclare, modified `isValidValueDescriptor()` as commented

File: internal/core/data/reading.go [hunk 3]
```diff
@@ -72,10 +73,11 @@ func readingHandler(w http.ResponseWriter, r *http.Request) {
 				return
 			}
 
-			valid, err := isValidValueDescriptor(vd, reading)
-			if !valid {
-				http.Error(w, "Validation failed", http.StatusConflict)
-				LoggingClient.Error("Validation failed")
+			err = isValidValueDescriptor(vd, reading)
+			if err != nil {
+				err = errors.NewErrValueDescriptorInvalid(vd.Name, err)
```
Comment (feclare) [inline internal/core/data/reading.go]: Maybe, instead of creating the ErrValueDescriptorInvalid here that isValidValueDescriptor return it directly?

Comment (jduranf) [reply internal/core/data/reading.go]: Done!

File: pkg/models/subscription.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ import (
  * Subscription struct
  */
 type Subscription struct {
-	BaseObject                                   `bson:",inline"`
+	BaseObject           `bson:",inline"`
```
Comment (drasko) [inline pkg/models/subscription.go]: Looks confusing, needs indenting and alignment to other fields.

Comment (jduranf) [reply pkg/models/subscription.go]: I agree with you, looks confusing, but otherwise gofmt fails... also there are several other structs (like Addressable or Command) with the same "confusing" structure, so I think it's better to not modify again this line.

Comment (drasko) [reply pkg/models/subscription.go]: OK, valid points.

**Comment by drasko** [comment]
@jduranf please update branch and let's merge this baby ;)
